### DISCUSSION TITLE
Issue with Setting Editable with DataGrid

### DISCUSSION THREAD

**Discussion by JMoodyUMC** [Megabit/Blazorise#3421] [Q&A] [answered]
I  have a DataGrid where I keep display the wage history of employees. In that grid, I have a DataGridNumericColumn where the Editable flag is based on the value in the DataGridCheckColumn. I cannot get the DataGridNumericColumn to turn editable when I set the DataGridCheckColumn to true if I am trying to edit a row.

```
<DataGrid @ref="dataGrid" TItem="WageHistory" Data="@employeeService.Employee.WageHistories" Editable="true" EditMode="DataGridEditMode.Inline" UseInternalEditing="true" @bind-SelectedRow="@selectedWage">
    <DataGridColumns>
        <DataGridCheckColumn TItem="WageHistory" Field="@nameof(WageHistory.RedCircleFlag)" Caption="Red Circle Flag" Editable="true" Width="80px" />
        <DataGridNumericColumn TItem="WageHistory" Field="@nameof(WageHistory.PackageRate)" Caption="Package Rate" Editable="@IsWageEditable()" Width="80px" />
        <DataGridCommandColumn TItem="WageHistory" DisplayOrder="10" DeleteCommandAllowed="false">
            <EditCommandTemplate>
                <Button Color="Color.Primary" Clicked="@(async () => { await dataGrid.Select(context.Item); await context.Clicked.InvokeAsync(); })" Size="Size.ExtraSmall" Margin="Margin.Is1.FromRight">Edit</Button>
            </EditCommandTemplate>
        </DataGridCommandColumn>
    </DataGridColumns>
</DataGrid>
```

```
@code {
    DataGrid<WageHistory> dataGrid;
    WageHistory selectedWage;

    private Boolean IsWageEditable()
    {
        if (selectedWage == null)
        {
            return true;
        }
        else
        {
            return selectedWage.RedCircleFlag;
        }
    }
}
```

Should I convert the DataGridCheckColumn to a DataGridColumn with a EditTemplate and DisplayTemplate so I can set the Checked and CheckedChanged to set a variable that can be used to set the editable flag on the numeric column?

Thanks,
Jill

**Comment by stsrki** [comment, marked as answer]
I don't think it would work through the `Editable` flag since the model would probably have to be in sync with Blazor rendering state, and that can be difficult sometimes.

The best way I think would be to use `EditTemplate` and `DisplayTemplate` and then based on `IsWageEditable()` you can render theirs content or make them readonly/disabled.
